File: StackLinkedListPy/StackLinkedList.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None
    
class Stack:
    def __init__(self):
        self.top = None
    
    def isEmpty(self):
        if self.top == None:
            return True
        return False
    
    def push(self,data):
        newNode = Node(data)
        if self.top == None:
            self.top = Node(data)
        newNode.next = self.top
        self.top = newNode

    def pop(self):
        if self.top ==None:
            print("Stack is empty!")
        self.top = self.top.next

    def printStack(self):
        curr = self.top
        while curr.next is not None:
            print(curr.data)
            curr = curr.next

    def peek(self):
        if self.top == None:
            print("Stack is empty")
            return
        return print("Top element is:",self.top.data)

Stack1 = Stack()
Stack1.push(5)
Stack1.push(2)
Stack1.push(222)
Stack1.push(20)
Stack1.push(50)
# Stack1.pop()
Stack1.printStack()
Stack1.peek()

File: stackLinkedList/src/main/java/org/example/StackLinkedList.java
package org.example;

import java.util.Iterator;
import java.util.NoSuchElementException;
import java.util.Spliterator;
import java.util.function.Consumer;

public class StackLinkedList<Item> implements Stack<Item>{
    private Node top;
    private int size;


    private class Node{
        Item item;
        Node next;
    }
    public StackLinkedList(){
        top = null;
        size = 0;
    }

    @Override
    public Item pop() {
        Item item = top.item;
        top = top.next;
        size--;
        return item;
    }

    @Override
    public void push(Item item) {
        Node oldTop = top;
        top = new Node();
        top.item = item;
        top.next = oldTop;
        size++;
    }

    @Override
    public boolean isEmpty() {
        return top == null;
    }

    @Override
    public int size() {
        return size;
    }
    public Iterator<Item> iterator(){
        return new LinkedListStackIterator();
    }
    private class LinkedListStackIterator implements Iterator<Item>{
        private int i = size;
        private Node first = top;
        @Override
        public boolean hasNext() {
            return i > 0;
        }

        @Override
        public Item next() {
            Item item = first.item;
            first = first.next;
            i--;
            return item;
        }
    }
}